internal/repos/jvm_packages.go

```diff
@@ -59,6 +69,44 @@ func (s *JVMPackagesSource) listDependentRepos(ctx context.Context, results chan
 			Repo:   repo,
 		}
 	}
+	if err != nil {
+		results <- SourceResult{Err: err}
+		return
+	}
+
+	dbDeps, err := s.dbStore.GetJVMDependencyRepos(ctx)
+	if err != nil {
+		results <- SourceResult{Err: err}
+		return
+	}
+
+	for _, dep := range dbDeps {
+		parsedModule, err := reposource.ParseMavenModule(dep.Identifier)
+		if err != nil {
+			log15.Warn("error parsing maven module", "error", err, "module", dep.Identifier)
+			continue
+		}
+		fullDependencyString := parsedModule.SortText() + ":" + dep.Version
```

Comment:
You can construct the dependency struct directly
```suggestion
		mavenDependency := MavenDependency{Module: parsedModule, Version: dep.Version}
```